File: solutions/scala/033-first-missing-positive.scala
/**
 * Problem 33: First Missing Positive (LeetCode 41)
 * Difficulty: Hard
 * Language: Scala
 */
def firstMissingPositive(nums: Array[Int]): Int = {
  val s = nums.filter(_ > 0).toSet
  Iterator.from(1).find(i => !s.contains(i)).get
}

File: solutions/rust/033-first-missing-positive.rs
/**
 * Problem 33: First Missing Positive (LeetCode 41)
 * Difficulty: Hard
 * Language: Rust
 */
use std::collections::HashSet;

pub fn first_missing_positive(nums: Vec<i32>) -> i32 {
    let s: HashSet<i32> = nums.into_iter().filter(|&n| n > 0).collect();
    (1..).find(|n| !s.contains(n)).unwrap()
}
